pythnet/message_buffer/programs/mock-cpi-caller/tests/program_test/mod.rs

```diff
@@ -0,0 +1,681 @@
+    }
+
+    pub fn default_allowed_programs() -> Vec<Pubkey> {
+        vec![MessageBufferTestContext::get_mock_cpi_auth()]
+    }
+
+
+    pub fn default_pyth_price_account() -> Pubkey {
+        Self::get_mock_pyth_price_account(Self::DEFAULT_TEST_PRICE_ID)
+    }
+
+    pub fn get_mock_pyth_price_account(id: u64) -> Pubkey {
+        let (mock_pyth_price_acct, _) = Pubkey::find_program_address(
+            &[b"pyth".as_ref(), b"price".as_ref(), &id.to_le_bytes()],
+            &::mock_cpi_caller::id(),
+        );
+        mock_pyth_price_acct
+    }
+
+    pub fn default_msg_buffer() -> (Pubkey, u8) {
+        find_msg_buffer_pda(
+            Self::get_mock_cpi_auth(),
+            Self::default_pyth_price_account(),
+        )
+    }
+
+
+    pub async fn process_ixs(
+        &mut self,
+        instructions: &[Instruction],
+        signers: Vec<&Keypair>,
+    ) -> anchor_lang::Result<()> {
+        let recent_blockhash = self.context.get_new_latest_blockhash().await.unwrap();
+
+        let mut transaction = Transaction::new_with_payer(instructions, Some(&self.payer.pubkey()));
+        transaction.partial_sign(&[&self.payer], recent_blockhash);
+        transaction.partial_sign(&signers, recent_blockhash);
+
+        let res = self
+            .context
+            .banks_client
+            .process_transaction(transaction)
+            .await;
+        match res {
+            Err(BanksClientError::TransactionError(TransactionError::InstructionError(
+                _,
+                InstructionError::Custom(error_code),
+            ))) => {
+                let e = Custom(error_code);
+                Err(e.into())
+            }
+            Err(_) => panic!("Unexpected error"),
+            Ok(_) => Ok(()),
+        }
+    }
+
+    pub async fn initialize(&mut self, admin: Keypair) -> Result<(Pubkey, u8)> {
+        let (whitelist_pda, whitelist_bump) = Pubkey::find_program_address(
+            &[b"message".as_ref(), b"whitelist".as_ref()],
+            &::message_buffer::id(),
+        );
+
+        self.admin = Some(admin.insecure_clone());
+        self.whitelist = Some(whitelist_pda);
+
+        let init_message_buffer_ix =
+            initialize_ix(admin.pubkey(), self.payer.pubkey(), whitelist_pda);
+
+        self.process_ixs(
+            &[init_message_buffer_ix],
+            vec![&self.admin.as_ref().unwrap().insecure_clone()],
+        )
+        .await
+        .unwrap();
+
+        Ok((whitelist_pda, whitelist_bump))
+    }
+
+    pub async fn fetch_whitelist(&mut self) -> Result<(u8, Pubkey, u32, Vec<Pubkey>)> {
+        let whitelist_account = self
+            .context
+            .banks_client
+            .get_account(self.whitelist())
+            .await
+            .unwrap();
+        assert!(whitelist_account.is_some());
+        let whitelist_account = whitelist_account.unwrap();
+        let account_data = whitelist_account.data();
+
+        deserialize_whitelist(account_data)
```

Comment:
It's weird to me that you needed to write this `deserialize_whitelist` when there should be an anchor function `Whitelist::try_deserialize` that does this

Reply:
I was writing these in a way that didn't use anchor so it could be more easily used by other components